internal/castai/castai.go

```diff
@@ -78,14 +80,18 @@ func NewDefaultRestyClient() *resty.Client {
 
 // NewDefaultDeltaHTTPClient configures a default http client used for sending delta requests. Delta requests use a
 // different client due to the need to access various low-level features of the http.Client.
-func NewDefaultDeltaHTTPClient() *http.Client {
+func NewDefaultDeltaHTTPClient(log logrus.FieldLogger) *http.Client {
 	return &http.Client{
 		Timeout:   sendDeltaReadTimeout,
-		Transport: createHTTPTransport(),
+		Transport: createHTTPTransport(log),
 	}
 }
 
-func createHTTPTransport() *http.Transport {
+func createHTTPTransport(log logrus.FieldLogger) *http.Transport {
+	tlsConfig, err := createTLSConfig()
+	if err != nil {
+		log.Errorf("creating TLS config: %v", err)
```

Comment:
What is the reason for not quitting the boot process if TLS config fails to get created? Shouldn't the agent go into a CrashLoop state instead?

Reply:
My reasoning was that agent can still proceed with default TLS configuration (with TLSClientConfig set to nil) and then CrashLoop if TLS negotiation still fails.

Reply:
But if user decided to configure TLS and it fails, why would it continue? I'd rather user configuration was validated and if invalid, then fail.

Reply:
Added agent fail if TLS config fails to be created with user provided cert.